zb-db/src/main/java/io/zeebe/db/impl/rocksdb/ZeebeRocksDbFactory.java

```diff
@@ -55,84 +66,196 @@ private ZeebeRocksDbFactory(
-          new DBOptions()
-              .setCreateMissingColumnFamilies(true)
-              .setErrorIfExists(false)
-              .setCreateIfMissing(true)
-              .setParanoidChecks(true);
+      final var dbOptions = createDefaultDbOptions(closeables);
       closeables.add(dbOptions);
 
-      db =
-          ZeebeTransactionDb.openTransactionalDb(
-              dbOptions,
-              dbDirectory.getAbsolutePath(),
-              columnFamilyDescriptors,
-              closeables,
-              columnFamilyTypeClass);
+      final var options = new Options(dbOptions, columnFamilyOptions);
+      closeables.add(options);
+
+      db = ZeebeTransactionDb.openTransactionalDb(options, pathName.getAbsolutePath(), closeables);
 
     } catch (final RocksDBException e) {
-      throw new RuntimeException("Unexpected error occurred trying to open the database", e);
+      CloseHelper.quietCloseAll(closeables);
+      throw new IllegalStateException("Unexpected error occurred trying to open the database", e);
     }
     return db;
   }
 
-  private List<ColumnFamilyDescriptor> createFamilyDescriptors(
-      final List<byte[]> columnFamilyNames, final ColumnFamilyOptions columnFamilyOptions) {
-    final List<ColumnFamilyDescriptor> columnFamilyDescriptors = new ArrayList<>();
+  private DBOptions createDefaultDbOptions(final List<AutoCloseable> closeables) {
+    final var statistics = new Statistics();
+    closeables.add(statistics);
+    statistics.setStatsLevel(StatsLevel.ALL);
 
-    if (columnFamilyNames != null && !columnFamilyNames.isEmpty()) {
-      for (final byte[] name : columnFamilyNames) {
-        final ColumnFamilyDescriptor columnFamilyDescriptor =
-            new ColumnFamilyDescriptor(name, columnFamilyOptions);
-        columnFamilyDescriptors.add(columnFamilyDescriptor);
-      }
-    }
-    return columnFamilyDescriptors;
+    return new DBOptions()
+        .setErrorIfExists(false)
+        .setCreateIfMissing(true)
+        .setParanoidChecks(true)
+        // 1 flush, 1 compaction
+        .setMaxBackgroundJobs(2)
+        // we only use the default CF
+        .setCreateMissingColumnFamilies(false)
+        // may not be necessary when WAL is disabled, but nevertheless recommended to avoid
+        // many small SST files
+        .setAvoidFlushDuringRecovery(true)
+        // fsync is called asynchronously once we have at least 4Mb
+        .setBytesPerSync(4 * 1024 * 1024L)
+        // limit the size of the manifest (logs all operations), otherwise it will grow
+        // unbounded
+        .setMaxManifestFileSize(256 * 1024 * 1024L)
+        // speeds up opening the DB
+        .setSkipStatsUpdateOnDbOpen(true)
+        // keep 1 hour of logs - completely arbitrary. we should keep what we think would be
+        // a good balance between useful for performance and small for replication
+        .setLogFileTimeToRoll(Duration.ofMinutes(30).toSeconds())
+        .setKeepLogFileNum(2)
```

Comment:
afaik these are just for the log file - yes, we should set `maxLogFileSize`, good catch :muscle:

Reply:
Java Doc
```java
  /**
   * Specifies the maximum size of a info log file. If the current log file
   * is larger than `max_log_file_size`, a new info log file will
   * be created.
   * If 0, all logs will be written to one log file.
   *
   * @param maxLogFileSize the maximum size of a info log file.
   * @return the instance of the current object.
   * @throws java.lang.IllegalArgumentException thrown on 32-Bit platforms
   *   while overflowing the underlying platform specific value.
   */
  T setMaxLogFileSize(long maxLogFileSize);

  /**
   * Returns the maximum size of a info log file. If the current log file
   * is larger than this size, a new info log file will be created.
   * If 0, all logs will be written to one log file.
   *
   * @return the maximum size of the info log file.
   */
  long maxLogFileSize();

  /**
   * Specifies the time interval for the info log file to roll (in seconds).
   * If specified with non-zero value, log file will be rolled
   * if it has been active longer than `log_file_time_to_roll`.
   * Default: 0 (disabled)
   *
   * @param logFileTimeToRoll the time interval in seconds.
   * @return the instance of the current object.
   * @throws java.lang.IllegalArgumentException thrown on 32-Bit platforms
   *   while overflowing the underlying platform specific value.
   */
  T setLogFileTimeToRoll(long logFileTimeToRoll);

  /**
   * Returns the time interval for the info log file to roll (in seconds).
   * If specified with non-zero value, log file will be rolled
   * if it has been active longer than `log_file_time_to_roll`.
   * Default: 0 (disabled)
   *
   * @return the time interval in seconds.
   */
  long logFileTimeToRoll();

  /**
   * Specifies the maximum number of info log files to be kept.
   * Default: 1000
   *
   * @param keepLogFileNum the maximum number of info log files to be kept.
   * @return the instance of the current object.
   * @throws java.lang.IllegalArgumentException thrown on 32-Bit platforms
   *   while overflowing the underlying platform specific value.
   */
  T setKeepLogFileNum(long keepLogFileNum);

  /**
   * Returns the maximum number of info log files to be kept.
   * Default: 1000
   *
   * @return the maximum number of info log files to be kept.
   */
  long keepLogFileNum();
```

https://github.com/facebook/rocksdb/blob/7c19d43883bdbb26b1cfceeeebeee983e7d5cf20/include/rocksdb/options.h#L601-L617
```cpp
  // Specify the maximal size of the info log file. If the log file
  // is larger than `max_log_file_size`, a new info log file will
  // be created.
  // If max_log_file_size == 0, all logs will be written to one
  // log file.
  size_t max_log_file_size = 0;

  // Time for the info log file to roll (in seconds).
  // If specified with non-zero value, log file will be rolled
  // if it has been active longer than `log_file_time_to_roll`.
  // Default: 0 (disabled)
  // Not supported in ROCKSDB_LITE mode!
  size_t log_file_time_to_roll = 0;

  // Maximal info log files to be kept.
  // Default: 1000
  size_t keep_log_file_num = 1000;
```

I'm not sure here. We set the log file to roll, so as far as I understand it will be enough, since we have not size set we use anyway only one file?

Reply:
If I check the long running benchmark we can see that we have two files. One old one and a current one. I assume after 30 min it marks the current as old and writes into a new one


```
root@zell-one-col-memory-zeebe-0:/usr/local/zeebe# ls -lah data/raft-partition/partitions/1/runtime/
total 117M
drwxr-xr-x 2 root root 4.0K Jan 28 06:25 .
drwxr-xr-x 5 root root 4.0K Jan 28 06:25 ..
-rw-r--r-- 1 root root  37M Jan 28 06:25 028919.log
-rw-r--r-- 1 root root 2.0M Jan 28 06:25 028921.sst
-rw-r--r-- 1 root root  21M Jan 28 06:25 028922.log
-rw-r--r-- 1 root root   16 Jan 26 07:04 CURRENT
-rw-r--r-- 1 root root   37 Jan 26 07:04 IDENTITY
-rw-r--r-- 1 root root    0 Jan 26 07:04 LOCK
-rw-r--r-- 1 root root 940K Jan 28 06:25 LOG
-rw-r--r-- 1 root root 2.9M Jan 28 06:15 LOG.old.1611814546976970
-rw-r--r-- 1 root root 1.6M Jan 28 06:25 MANIFEST-000003
-rw-r--r-- 1 root root 6.1K Jan 26 07:04 OPTIONS-000006
```